units/patch_intent.go

```diff
@@ -1076,8 +1090,13 @@ func (j *patchIntentProcessor) buildTriggerPatchDoc(patchDoc *patch.Patch) (*mod
 		}
 		for _, p := range parentPatch.Patches {
 			if p.ModuleName == "" {
+				moduleName := intent.ParentAsModule
+				if patchDoc.Triggers.SameBranchAsParent {
+					patchDoc.Githash = parentPatch.Githash
+					moduleName = ""
+				}
 				patchDoc.Patches = append(patchDoc.Patches, patch.ModulePatch{
-					ModuleName: intent.ParentAsModule,
+					ModuleName: moduleName,
```

Comment:
This will add the parent as a module for any child patches that have the same branch, when we were previously not doing that. Do we maybe want to avoid that?

Reply:
The lines above make it such that if the child patch uses the same repo/branch as the parent patch then `moduleName` will be the empty string, meaning those changes will apply to the base project of the patch. I can add a comment making this clearer.

So the parent wont be added as a module, but the parent's patched changes will apply to both the parent and child patches.

Reply:
I can't really think of a use case where someone would specifically want a child patch and a parent patch that have the same repo/branch. However, "the parent's patched changes will apply to both the parent and child patches" doesn't feel right, I don't think users expect that to be the case for child patches.

Reply:
I can't think of a use case where someone would want a child and parent patch with the same repo/branch. However, "the parent's patched changes will apply to both the parent and child patches" seems like it might break user expectations for child patches. I think users would expect the patch changes to only apply to the parent patch. And for the parent project's changes to be included only if the parent project is set up as a module in the patch trigger alias. 

I can't think of a use case where a parent and child with the same repo/branch would be needed. Are you aware of a use case like that? I wonder if that would indicate an error state or a misconfiguration.

Reply:
Sorry to have this thread going on for too long.

> I can't think of a use case where someone would want a child and parent patch with the same repo/branch.

I initially thought the same thing, but this use case is one of the reasons this ticket was created. It turns out mongodb-mongo-master sometimes creates child patches in the `mongodb-mongo-master-nightly` project ([example](https://spruce.mongodb.com/version/65576a9d32f417c7eef6ae38/downstream-projects?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC))

Reply:
Aah very interesting! 
Can you please add a comment to the code explaining this so if people touch that code down the line they know to keep that functionality?